src/main/java/com/commercetools/sync/services/impl/UnresolvedReferencesServiceImpl.java

```diff
@@ -17,23 +17,27 @@
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
 
 import static java.lang.String.format;
 import static java.util.stream.Collectors.toList;
+import static org.apache.commons.codec.digest.DigestUtils.sha1Hex;
 
 public class UnresolvedReferencesServiceImpl implements UnresolvedReferencesService {
 
-    private final ProductSyncOptions syncOptions;
-
-    private static final String SAVE_FAILED = "Failed to save CustomObject with key: '%s'. Reason: %s";
+    private static final String SAVE_FAILED = "Failed to save CustomObject with key: '%s' (hash of product key: '%s'). Reason: %s";
     private static final String DELETE_FAILED = "Failed to delete CustomObject with key: '%s'. Reason: %s";
-    private static final String CUSTOM_OBJECT_CONTAINER_KEY =
-        "commercetools-sync-java.UnresolvedReferencesService.productDrafts";
+    private static final String CUSTOM_OBJECT_CONTAINER_KEY = "commercetools-sync-java.UnresolvedReferencesService.productDrafts";
+    private final ProductSyncOptions syncOptions;
 
     public UnresolvedReferencesServiceImpl(@Nonnull final ProductSyncOptions baseSyncOptions) {
         this.syncOptions = baseSyncOptions;
     }
 
+    private String hash(@Nonnull final String customObjectKey) {
```

Comment:
Could you please add `@Nonnull` for the method return? so its easier for the caller.